### PR TITLE
UI tests coverage, fixes and updates

### PR DESCRIPTION
**Pull request by FJBDev**
- The fix for the `argLine` issue was inspired from [here](https://www.eclipse.org/lists/tycho-user/msg03519.html)

- [PoiView]: Replacing the deprecated [Observable ](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/Observable.html)pattern with [PropertyChangeListener approach](https://www.baeldung.com/java-observer-pattern)

- FIT SDK [21.89](https://forums.garmin.com/developer/fit-sdk/b/news-announcements/posts/fit-sdk-21-89-00-release)

- Migrated the app that allows for the cloud integration in MT from **Heroku** to [Cyclic](https://www.cyclic.sh/)

- Now uploading the full JaCoCo code coverage report for each successful build to be imported in Eclipse with [EclEmma](https://www.eclemma.org/)

- FIT Import: Fixed NPE when pauses data are missing (https://sourceforge.net/p/mytourbook/discussion/622811/thread/d52269ce04/)

### CODE REVIEW

File: bundles/net.tourbook.common/src/net/tourbook/common/util/XmlUtils.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import org.xml.sax.SAXException;
+
+public class XmlUtils {
+
+   public static SAXParser initializeParser() {
+
+      try {
+
+         final SAXParserFactory factory = SAXParserFactory.newInstance();
+         factory.setFeature("http://apache.org/xml/features/disallow-doctype-decl", true); //$NON-NLS-1$
```
Comment (wolfgang-ch) [inline bundles/net.tourbook.common/src/net/tourbook/common/util/XmlUtils.java:31]: It may be helpful to add a comment that this is fixing a security issue

Comment (FJBDev) [reply bundles/net.tourbook.common/src/net/tourbook/common/util/XmlUtils.java:31]: Commend added

**Comment by wolfgang-ch** [comment]
The lastest main is not merged into this PR

image

**Comment by wolfgang-ch** [comment]
> > The lastest main is not merged into this PR
> 
> I have merged it with your latest main, it should be good now.

Now the git compare was OK

image

image

**Comment by FJBDev** [comment]
> Now the git compare was OK

👍

**Pull request merged**